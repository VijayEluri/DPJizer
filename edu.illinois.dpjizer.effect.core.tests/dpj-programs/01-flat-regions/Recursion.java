/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package flatregion.recursion;

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class C {
	region r1;
	region r2;
	int a in r1;
	int b in r2;

	C() {}
	
	void m1() {
		a = 1;
		m2();
	}
	
	void m2() {
		b = 2;
		m1();
	}
}