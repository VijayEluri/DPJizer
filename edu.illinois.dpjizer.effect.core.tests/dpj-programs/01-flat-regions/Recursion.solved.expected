flatregion.recursion.C.C() pure
flatregion.recursion.C.m1() writes Root : flatregion.recursion.C.r1, Root : flatregion.recursion.C.r2
flatregion.recursion.C.m2() writes Root : flatregion.recursion.C.r1, Root : flatregion.recursion.C.r2
